test/test.c

```diff
@@ -666,21 +669,84 @@ bool testLoadScratchpad()
 
     if (status.scrat_len != file_size)
     {
-        LOGE("Scratchpad is not loaded correctly (wrong size) %d vs %d\n",
+        LOGE("Scratchpad is not uploaded correctly (wrong size) %d vs %d\n",
              status.scrat_len,
              file_size);
         return false;
     }
 
     if (status.scrat_seq_number != SEQ_NUMBER)
     {
-        LOGE("Wrong seq number after loading a scratchpad image \n");
+        LOGE("Wrong seq number after uploading a scratchpad image \n");
         return false;
     }
 
     return true;
 }
 
+bool testDownloadScratchpad()
+{
+    FILE * fp;
+    uint32_t scratchpad_size;
+    long read = 0;
+    const char * filename = OTAP_DOWNLOAD_FILE_PATH;
+    uint8_t block[BLOCK_SIZE];
+
+    fp = fopen(filename, "wb");
+    if (fp == NULL)
+    {
+        LOGE("Cannot open file %s. Please update OTAP_DOWNLOAD_FILE_PATH to a "
+             "valid path\n",
+             filename);
+        return false;
+    }
+
+    /* Get size of scratchpad */
+    if (WPC_get_scratchpad_size(&scratchpad_size) != APP_RES_OK)
+    {
+        LOGE("Cannot read scratchpad size\n");
+        return false;
+    }
+
+    LOGI("Downloading otap file of %d bytes\n", scratchpad_size);
+
+    /* Receive scratchpad image block by block */
+    LOGI("Start receiving otap (it may take up to 1 minute)\n");
+    uint32_t remaining = scratchpad_size;
+    while (remaining > 0)
+    {
+        uint8_t block_size = (remaining > BLOCK_SIZE) ? BLOCK_SIZE : remaining;
+
+        /* Receive the block */
+        if (WPC_download_local_scratchpad(block_size, block, read) != APP_RES_OK)
+        {
+            LOGE("Cannot download scratchpad block\n");
+            break;
+        }
+
+        size_t written = fwrite(block, 1, block_size, fp);
+        if (written != block_size)
+        {
+            LOGE("Error while writing file asked = %d read = %d\n", block_size, written);
+            break;
+        }
+
+        read += block_size;
+        remaining -= block_size;
+    }
+
+    fclose(fp);
+
+    if (remaining > 0)
+    {
+        return false;
+    }
+
+    LOGE("Please compare files %s and %s\n", OTAP_UPLOAD_FILE_PATH, OTAP_DOWNLOAD_FILE_PATH);
```

Comment:
LOGW (or LOGI) should be more appropriated

Reply:
Indeed. Fixed.